storage/include/storage/storage_metrics.h

```diff
@@ -14,41 +15,67 @@
 namespace concord {
 namespace storage {
 
+class StorageMetrics {
+ public:
+  virtual void updateMetrics() = 0;
+};
+
+class StorageMetricsUpdator {
+  StorageMetrics& storage_metrics_;
+  std::thread t;
+  std::chrono::milliseconds metrics_update_interval_ = std::chrono::milliseconds(100);  // TODO: read from configuration
+  std::atomic_bool active = false;
+
+ public:
+  StorageMetricsUpdator(StorageMetrics& storage_metrics) : storage_metrics_(storage_metrics) {}
+
+  void close() {
+    if (!active) return;
+    active = false;
+    t.join();
+  }
+  void start() {
+    active = true;
+    t = std::thread([&] {
+      while (active) {
+        storage_metrics_.updateMetrics();
```

Comment:
Can `updateMetrics()` throw? We can maybe be explicit about whether that can happen and, if yes, what is the desired behaviour.